src/Components/Form/Form.js

```diff
@@ -1,154 +1,243 @@
+        return a > b ? -1 : 1;
+      });
+      return filteredSet.slice(0, -1);
     }
-    return this.state.allAnswers[0]
-  }
+    return this.state.allAnswers[0];
+  };
 
   determinePrompt = (index, data) => {
     return (
-        <article className='form-container'>
-          <h2 className='question'>{data[index].question}</h2>
-          <div className='possible-answers'>
-            {data[index].choices.map((choice, i) => {
-              return (
-                <div className='choice'>
-                  <h3
-                    key={i}
-                    id={data[index].answerType}
-                    onClick={(e) => {
-                      this.updateCurrentAnswers(e, data[index])
-                      }
-                    }
-                    value={choice} 
-                    className='option'>{choice}
-                  </h3>
-                </div>
-                )
-            })}
-          </div>
-        </article>
-      )
-  }
+      <article className="form-container">
+        <h2 className="question">{data[index].question}</h2>
+        <div className="possible-answers">
+          {data[index].choices.map((choice, i) => {
+            return (
+              <div className="choice">
+                <h3
+                  key={i}
+                  id={data[index].answerType}
+                  onClick={(e) => {
+                    this.updateCurrentAnswers(e, data[index]);
+                  }}
+                  value={choice}
+                  className="option"
+                >
+                  {choice}
+                </h3>
+              </div>
+            );
+          })}
+        </div>
+      </article>
+    );
+  };
 
   showQuestion = () => {
     if (!this.props.activities.length && !this.state.allAnswers.length) {
       return this.determinePrompt(0, this.state.prompts);
     }
     if (this.state.questionsPerActivity.length) {
-      let unansweredSet = this.state.questionsPerActivity.find(set => {
-        return this.state.allAnswers[0][this.state.allAnswers.length - 1] === set.activity
-      })
+      let unansweredSet = this.state.questionsPerActivity.find((set) => {
+        return (
+          this.state.allAnswers[0][this.state.allAnswers.length - 1] ===
+          set.activity
+        );
+      });
       return unansweredSet.questions.map((question, i) => {
-        return this.determinePrompt(i, unansweredSet.questions)
+        return this.determinePrompt(i, unansweredSet.questions);
+      });
+    }
+  };
+
+  handleSubmission = (event) => {
+    event.preventDefault()
+    let index = this.state.allAnswers.length - 1;
+    if (
+      this.state.questionsPerActivity[index].questions.every((question) => {
+        return this.state.currentAnswers.some((answer) =>
+          question.choices.includes(answer)
+        );
       })
+    ) {
+      this.props.determineRandomActivity();
+    } else {
+      alert("Please answer all questions");
     }
-  }
+  };
```

Comment:
New logic for submit button to make sure the user has answered all the questions

Logic is a little difficult to read, but essentially the `every` iterator is assuring every question of the current activity has at least `some` of the current answers... if that makes sense.... I'll try to explain better next time we meet. But it does work!

Reply:
Makes sense to me, this verifies that any of the choices exist in that answers array.  Very sound logic.

Reply:
My understanding is that is checking `every` array to see if it contains an answer correct?